### PR TITLE
[BPK-4252] Add MarkdownPage and migrate status page to use it

### PR DESCRIPTION
**Pull request by shaundon**
Proof of concept of `MarkdownPage`, with the status page migrated to use it.

## Answers to questions you might have 

**Why are `title` and `subtitle` props instead of just coming from the markdown?**
We need to know the title so we can set `window.title`. I tried pulling it out automatically but it's hacky so I decided to be explicit instead. This way also allows us to put that nice white area at the top with the title and subtitle. When we expand this PoC to do some snazzy automatic generation stuff, we should be able to pull these out of front matter or something to set them, so it should scale.

**The background alternating colours don't work with these**
Nope. YAGNI.

Screen Shot 2020-10-26 at 16 41 43

### CODE REVIEW

File: docs/src/components/MarkdownPage/MarkdownPage.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import Heading from '../Heading';
+import IntroBlurb from '../IntroBlurb';
+
+import STYLES from './MarkdownPage.scss';
+
+const getClassName = cssModules(STYLES);
+
+type Props = {
+  title: string,
+  subtitle: ?string,
```
Comment (georgegillams) [inline docs/src/components/MarkdownPage/MarkdownPage.js:36]: I don't think this should be called `subtitle`, because it's a paragraph

Comment (shaundon) [reply docs/src/components/MarkdownPage/MarkdownPage.js:36]: There's nothing that says subtitles are a single line. This usage meets the dictionary definition almost perfectly:

Screen Shot 2020-10-26 at 16 59 32

Comment (georgegillams) [reply docs/src/components/MarkdownPage/MarkdownPage.js:36]: Actually that makes sense 😂

**Comment by georgegillams** [comment]
The headings look a bit big. They're as big as the main one at the top 🤔 
Are we planing to lose the alternating section colouring? Seems it would make every page look quite bland

File: docs/src/components/MarkdownPage/MarkdownPage.js (hunk 1 above)
Comment (shaundon) [reply docs/src/components/MarkdownPage/MarkdownPage.js:36]: You know me and grammar 😛

**Comment by shaundon** [comment]
> The headings look a bit big. They're as big as the main one at the top 🤔

They're the same size as on non-markdown pages

> Are we planing to lose the alternating section colouring? Seems it would make every page look quite bland

Yep, this is addressed in the PR description. The alternating sections are nice but they'd be a nightmare to implement when the page is being automatically generated from markdown, so I'd rather lose them than hack them back in.

**Comment by shaundon** [comment]
> Are we planing to lose the alternating section colouring? Seems it would make every page look quite bland

Another way I think of this is that this is first-and-foremost a documentation site, so the content is king. We should make it look good but it's a secondary concern compared to ease of use (both for users and us maintaining it).

Tbh I want to rip a few things out of it, those massive images that push all the content below the fold are ridiculous 🙈

**Comment by georgegillams** [comment]
Oh yeah they are. Everything just looks a bit tight and garish though 🤔

**Comment by shaundon** [comment]
> Everything just looks a bit tight and garish though

In the nicest way possible, I just don't think it matters. Not for a PoC anyway. We can look at increasing the margin around headings later if it's really a problem.

Also bear in mind this is a strange-looking page compared to most others where the headings will be more spaced out with content.

**Comment by georgegillams** [comment]
As for the massive images, as far as I'm concerned we can kill them 🎉

**Comment by shaundon** [comment]
@georgegillams I feel like I've addressed your questions — is this ok to approve or do you still have concerns?

**Review by georgegillams — APPROVED** [review]

**Pull request merged**